ios/FluentUI/Tooltip/TooltipView.swift

```diff
@@ -157,23 +237,188 @@ class TooltipView: UIView {
+        if arrowDirection.isVertical {
+            idealPosition = sourcePointInWindow.x - tooltipRect.minX - arrowWidth / 2
+            maxPosition = tooltipRect.width - arrowMargin - arrowWidth
+        } else {
+            idealPosition = sourcePointInWindow.y - tooltipRect.minY - arrowWidth / 2
+            maxPosition = tooltipRect.height - arrowMargin - arrowWidth
+        }
+        return max(minPosition, min(idealPosition, maxPosition))
     }
 
-    override var accessibilityHint: String? {
-        get { return "Accessibility.Dismiss.Hint".localized }
-        set { }
+    private var idealTooltipOrigin: CGPoint {
+        switch arrowDirection {
+        case .up:
+            return CGPoint(x: sourcePointInWindow.x - tooltipSize.width / 2, y: sourcePointInWindow.y)
+        case .down:
+            return CGPoint(x: sourcePointInWindow.x - tooltipSize.width / 2, y: sourcePointInWindow.y - tooltipSize.height)
+        case .left:
+            return CGPoint(x: sourcePointInWindow.x, y: sourcePointInWindow.y - tooltipSize.height / 2)
+        case .right:
+            return CGPoint(x: sourcePointInWindow.x - tooltipSize.width, y: sourcePointInWindow.y - tooltipSize.height / 2)
+        }
     }
+
+    private var sourcePointInAnchorView: CGPoint {
+        switch arrowDirection {
+        case .up:
+            return CGPoint(x: anchorView.frame.width / 2, y: anchorView.frame.height)
+        case .down:
+            return CGPoint(x: anchorView.frame.width / 2, y: 0)
+        case .right:
+            return CGPoint(x: 0, y: anchorView.frame.height / 2)
+        case .left:
+            return CGPoint(x: anchorView.frame.width, y: anchorView.frame.height / 2)
+        }
+    }
+
+    private var sourcePointInWindow: CGPoint {
+        return anchorView.convert(sourcePointInAnchorView, to: window)
+    }
+
+    private var boundingRect: CGRect {
+        let screenMargin = TooltipTokenSet.screenMargin
+        guard let window = anchorView.window else {
+            preconditionFailure("Can't find anchorView's window")
+        }
+
+        return window.bounds.inset(by: window.safeAreaInsets).inset(by: UIEdgeInsets(top: screenMargin,
+                                                                                     left: screenMargin,
+                                                                                     bottom: screenMargin,
+                                                                                     right: screenMargin))
+    }
+
+    private func anchorViewInset(for arrowDirection: Tooltip.ArrowDirection) -> UIEdgeInsets {
+        var inset = UIEdgeInsets.zero
+        let anchorViewFrame = anchorView.convert(anchorView.bounds, to: window)
+        switch arrowDirection {
+        case .up:
+            inset.top = max(anchorViewFrame.maxY - boundingRect.minY, 0)
+        case .down:
+            inset.bottom = max(boundingRect.maxY - anchorViewFrame.minY, 0)
+        case .left:
```

Comment:
q: Does RTL look ok? I ask because I see that UIEdgeInsets still use left/right instead of leading/trailing

Reply:
yup, RTL is all good!